File: src/java/org/elastos/hive/endpoint/NodeVersion.java
package org.elastos.hive.endpoint;

import com.google.gson.annotations.SerializedName;

public class NodeVersion {
	@SerializedName("major")
	long major;

	@SerializedName("minor")
	long minor;

	@SerializedName("patch")
	long patch;

	public long major() {
		return major;
	}

	public long minor() {
		return minor;
	}

	public long patch() {
		return patch;
	}

	@Override
	public String toString() {
		return String.format("%d.%d.%d", major, minor, patch);
	}
}

File: src/domain/nodeversion.ts
export class NodeVersion {
	private major: number;
	private minor: number;
	private patch: number;

	constructor(major: number, minor: number, patch: number) {
		this.major = major;
		this.minor = minor;
		this.patch = patch;
	}

	public getMajor(): number {
		return this.major;
	}

	public getMinor(): number {
		return this.minor;
	}

	public getPatch(): number {
		return this.patch;
	}

	public toString(): string {
		let version : number[] = [];

		this.major && version.push(this.major);
		this.minor && version.push(this.minor);
		this.patch && version.push(this.patch);

		return version.join(".");
	}
}
